src/shared/sleep-config.c

```diff
@@ -259,6 +265,52 @@ static bool enough_memory_for_hibernation(void) {
         return r;
 }
 
+static int parse_proc_cmdline_item(const char *key, const char *value) {
+
+        assert(key);
+
+        if (streq(key, "resume") && value) {
+                free(arg_resume_dev);
+                arg_resume_dev = fstab_node_to_udev_node(value);
+                if (!arg_resume_dev) {
+                        return log_oom();
+                }
+        }
+
+        return 0;
+
+}
+
+static bool resume_passed_to_kernel(void) {
+        int r = 0;
+        struct stat rd;
+
+        /* If not using the systemd hibernate generator we don't care about the cmdline */
+        if (!validate_hibernate_generator)
+                return true;
```

Comment:
This is not all or nothing. While initrd may already contain resume device information, kernel command line overrides it. So logic probably should be - *if* resume= is present, validate it.